docs/design/tool-prefix-matching.md

```diff
@@ -0,0 +1,53 @@
+## Tool Prefix Matching
+
+### Problem
+
+MCP Gateway routes tool calls to backend MCP servers based on tool name prefixes. Each MCPServer resource defines a `toolPrefix` field that the router uses to match incoming tool calls against registered backends.
+
+Current routing implementation checks MCP servers sequentially (in undefined order) until finding a prefix match. This creates ambiguity when:
+
+- Multiple MCP servers share the same `toolPrefix`
+- An MCP server has an empty `toolPrefix` (matches all tool names)
+
+An empty prefix acts as a wildcard, matching any tool name. If checked before prefixed servers, it incorrectly captures tool calls intended for specific backends. This breaks the prefix-based routing model.
+
+**Example Failure:**
+```
+MCPServer A: toolPrefix="" (empty)
+MCPServer B: toolPrefix="test_"
+
+Tool call: "test_tool_from_B"
+Current behavior: Routes to A (wrong)
+Expected behavior: Routes to B (correct)
+```
+
+### Solution
+
+Enforce single empty-prefix server and prioritize prefix matching:
+
+1. **Validation**: Only one MCPServer with an empty toolPrefix is allowed per namespace. Additional MCPServers with empty prefix fail validation with clear error message.
```

Comment:
what is 'namespace' in this context?
A kubernetes namespace?

Reply:
Why is only one empty toolPrefix allowed per namespace?

I almost **never** use a prefix, because I don't want to rewrite my agents to use different names.

Reply:
The current implementation uses the tool prefix to determine which server to route a tool call to.
If there's more than 1 mcp server without a tool prefix, there's no way to know which server a tool call should be routed to.

> it is possible to have more MCP Gateways deployed on cluster so there could be multiple MCPServer CRs having empty prefix as long as each is registered to a different MCP Gateway

Understood. That's a good point for consideration in #436 
As things stand, only 1 gateway instance is supported.